### PR TITLE
updates to porting guide

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

porting duige
##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 1]
```diff
@@ -158,7 +158,36 @@ desired.
...
+  The previous host_list, script, yaml and ini plugins still work as they did, the auto plugin is now the last one we attempt to use.
+  If you had customized the enabled plugins you might want to revise the setting to include the new auto plugin.
+
+Shell
+-----
+
+Shell plugins have been migrated to the new plugin configuration framework, as such it is now possible to customize more settings and previouslly 'global' settings can now also be overriden using host specific variables.
+
+For example, ``system_temps`` is a new setting that allows you to control what Ansible will consider a 'system temporary dir', previouslly it was hardcoded to '/tmp', now it defaults to ``[ '/var/tmp', '/tmp']``.
+This is used when using privilege escalation to a non administrative user, some systems cannot use `/tmp` for this purpose, this allows you to customize it globaly or for those systems.
```
Comment (samdoran) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Double backticks around '/tmp' and \`/tmp\`.

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 2]
```diff
@@ -158,7 +158,36 @@ desired.
...
+
+Another setting of interest is ``admin_users`` which allows you to specify a list of users to be considerd 'administrators', previouslly this was hardcoded to ``root``, now it defaults to ``[root, toor, admin]``.  This information is used when choosing between your ``remote_temp`` and a ``system_temps`` directory as mentioned above.
+
+For a full list, check the shell plugin you are using, the default shell plugin is ``sh``.
```
Comment (samdoran) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Separate sentences: 

For a full list, check the shell plugin you are using. The default shell plugin is ``sh``.

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 3]
```diff
@@ -158,7 +158,36 @@ desired.
...
+  Moving to the interal cache will allow you to use Ansible's existing cache refresh/invalidation mechanisms.
+
+* A new 'auto' plugin, enabled by default, that can automatically detect the correct plugin to use IF that plugin is using our 'common YAML configuration format'.
+  The previous host_list, script, yaml and ini plugins still work as they did, the auto plugin is now the last one we attempt to use.
+  If you had customized the enabled plugins you might want to revise the setting to include the new auto plugin.
+
+Shell
+-----
+
+Shell plugins have been migrated to the new plugin configuration framework, as such it is now possible to customize more settings and previouslly 'global' settings can now also be overriden using host specific variables.
```
Comment (samdoran) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Separate sentences:

"...plugin configuration framework. As such..."

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 4]
```diff
@@ -158,7 +158,36 @@ desired.
...
 
-No notable changes.
+As a developer you can now use 'doc fragments' for common configuration options on plugin types that support the new plugin configuration system.
+
+Inventory
+---------
+
+Inventory plugins have been fine tuned, and we have started to add some common features:
+
+* The ability to use a cache plugin to avoid costly API/DB queries, this is disabled by default.
```
Comment (samdoran) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Drop ", this"

"...costly API/DB queries is disabled by default."

**Review by dharmabumstead — APPROVED** [review]
OK with edits.

**Comment by nitzmahone** [comment]
cherry-picked to stable-2.5 for 2.5.0b2

**Pull request merged**